mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoCRUDController.java

```diff
@@ -815,12 +818,17 @@ private void createUpdateEntityInfoIndexes(EntityInfo ei, Metadata md) {
                     newIndex.put(Translator.translatePath(field), p.isDesc() ? -1 : 1);
                 }
                 BasicDBObject options = new BasicDBObject("unique", index.isUnique());
-                // if index is unique also make it a sparse index, so we can have non-required unique fields
-                options.append("sparse", index.isUnique());
+                // if index is unique and non-partial, also make it a sparse index, so we can have non-required unique fields
+                options.append("sparse", index.isUnique() && !index.getProperties().containsKey(PARTIAL_FILTER_EXPRESSION_OPTION_NAME));
                 if (index.getName() != null && index.getName().trim().length() > 0) {
                     options.append("name", index.getName().trim());
                 }
                 options.append("background", true);
+                // partial index
+                if (index.getProperties().containsKey(PARTIAL_FILTER_EXPRESSION_OPTION_NAME)) {
+                    DBObject filter = (DBObject) JSON.parse(index.getProperties().get(PARTIAL_FILTER_EXPRESSION_OPTION_NAME).toString());
```

Comment:
The property is  "partialFilterExpression": "{ \"field3\": { \"$gt\": 5 } }", which is a string.

I'd much rather this be: "partialFilterExpression": { "field3": { "$gt": 5 } }

When you do this, the property value will be a map containing field3 key, with value being a map containing $gt as key, and 5 as value.